### PR TITLE
webRequest StreamFilter ondata object property clarification

### LINKED ISSUE #18601: The documentation should link to a doc about the event structure
### MDN URL

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/StreamFilter/ondata

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

There's no doc about the event structure except for 'The handler is passed an event object containing a data property'.
Is this the only property ? 

### What did you expect to see?

All docs about event handling webRequest.StreamFilter.onXXX should have a link to the event structure if it matters.
If it only contains a data property, then documentation should emphasize that data is the one and only property

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/mozilla/add-ons/webextensions/api/webrequest/streamfilter/ondata`
* MDN URL: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/StreamFilter/ondata
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/mozilla/add-ons/webextensions/api/webrequest/streamfilter/ondata/index.md
* Last commit: https://github.com/mdn/content/commit/fd9019f85fb1696c7d814735cc1cd051c4901585
* Document last modified: 2022-02-11T15:51:39.000Z

</details>

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Fixes [#18601](https://github.com/mdn/content/issues/18601) by providing clarification about the content of the `event` object.

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/streamfilter/ondata/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: webextensions.api.webRequest.StreamFilter.ondata
 
 {{AddonSidebar()}}
 
-An event handler that is called repeatedly when response data is available. The handler is passed an `event` object containing a `data` property, which includes a chunk of the response data as an {{jsxref("ArrayBuffer")}}.
+An event handler called repeatedly when response data is available. The handler is passed an [`event` interface object](/en-US/docs/web/api/event). However, the only reliable content in that object is the `data` property, which includes a chunk of the response data as an {{jsxref("ArrayBuffer")}}.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/webrequest/streamfilter/ondata/index.md:10]:
```suggestion
An event handler called repeatedly when response data is available. The handler is passed an [`Event` object](/en-US/docs/web/api/event) with a `data` property, which includes a chunk of the response data as an {{jsxref("ArrayBuffer")}}.
```

My statement about reliability was just commentary, doesn't need to be called out.

**Review by Rob--W — CHANGES_REQUESTED** [review]
StreamFilter has more events beyond ondata. Can you apply similar changes consistently across all events?

**Comment by rebloor** [comment]
> StreamFilter has more events beyond ondata. Can you apply similar changes consistently across all events?

@Rob--W but the other three events (onstart, onstop,  and onerror) make no mention of them receiving an `event` object

**Review by Rob--W — APPROVED** [review]
Fair, thanks for improving this one.

**Pull request merged**